### PR TITLE
Validate shaders in Travis. Hope this will fix issue #187.

### PR DESCRIPTION
**Pull request by vab9**
Not 100% if this will work. The shaders throw erros with glslangValidator on osx as it is, but that might mean nothing for Travis.

I'm also not sure if caching the validator is a good idea or not...

### CODE REVIEW

File: ci/install-glslangValidator.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -e
+
+echo "need to install glslangValidator..."
+mkdir validator/
+cd validator/
+wget "https://cvs.khronos.org/svn/repos/ogl/trunk/ecosystem/public/sdk/tools/glslang/Install/Linux/glslangValidator"
```
Comment (jonas-schievink) [inline ci/install-glslangValidator.sh]: Oh that's neat! I feared we'd have to compile it on our own.

File: ci/validate-shaders.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -o errexit -o nounset
+
+echo ""
+echo "=== Validating Shaders =============="
+
+SHADER_DIR=`find $PWD -name shader`
```
Comment (jonas-schievink) [inline ci/validate-shaders.sh]: I'd prefer to scan for shader files in the whole tree, like `check-basic-style.sh` does. You can even use the same pattern: `FILES='.+\.\(rs\|vert\|tesc\|tese\|geom\|frag\|comp\)'` (Note: our tessellation shaders have a wrong suffix for some reason)

File: ci/validate-shaders.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+trap "rm $ERROR_TMP; exit" SIGHUP SIGINT SIGTERM
+
+# Make sure only shaders are in client/shader/ for this to work
+for shader in $SHADER_DIR/*; do
+    echo "Validating: `basename $shader` ..."
+    if ! glslangValidator -s $shader > /dev/null;
+        then
+        echo "=== ERRORS in `basename $shader` =============" >> $ERROR_TMP
+        glslangValidator $shader | grep "ERROR" >> $ERROR_TMP
+        echo "" >> $ERROR_TMP
```
Comment (jonas-schievink) [inline ci/validate-shaders.sh]: Why the indirection through the temporary file? Theoretically you can just let `glslangValidator` output to stdout/stderr as normal and set an error flag if it fails. Take a look at the `check-basic-style.sh` script, it works similarly.

**Comment by jonas-schievink** [comment]
Travis is apparently running Ubuntu 12.04, not sure how easy it will be to get the validator to run.

12.04.......

**Comment by jonas-schievink** [comment]
By the way, all the `#N` comments are referencing random unrelated issues and PRs

**Comment by jonas-schievink** [comment]
The glslang project itself seems to explicitly enable Ubuntu 14.04 LTS: https://github.com/KhronosGroup/glslang/blob/master/.travis.yml

Sadly, this would increase CI startup times by quite a bit (as we'll no longer be using the fast container-based infrastructure).

**Comment by vab9** [comment]
Damn, ok! It was worth a try... Travis-CI is pretty cool though

**Pull request closed without merging**